pkg/kube/misc.go

```diff
@@ -66,15 +72,19 @@ func GetServiceInterfaceTarget(targetType string, targetName string, deducePort
 	} else if targetType == "pods" {
 		return nil, fmt.Errorf("VAN service interfaces for pods not yet implemented")
 	} else if targetType == "service" {
+		targetSvc, err := GetServiceTarget(targetName, namespace, cli)
```

Comment:
I don't think we want to do this for a target of type service. This would now fail if the service doesn't exist yet, which it would not do before. It also requires read permission for the service if it is in another namespace or else it will fail (though that is true if deduceports is true already).

But I think to begine with, its not clear that there is any value in taking the value of PublishNotReadyAddresses from the service here.

Reply:
You are right, the value PublishNotReadyAddresses is related to services with statefultsets, probably not useful in this case. Thanks!